### PR TITLE
enhance compatibility and add a error handle

### PR DESCRIPTION
**Pull request by youyou-pm10**
modify tmp_dir to fit windows,trans encofing to utf-8 and add a check to avoid unsanitizer path of image
修改tmp_dir路径以适应windows不能创建多级目录，修改编码方式以适应中文，增加检查防止md导入语雀又导出带来的图片路径错误

### CODE REVIEW

**Comment by alswl** [comment]
感谢你的贡献，请问能否将截图删除，换成文字表述，这样就不用管理图片了，也便于其他朋友检索。

**Comment by youyou-pm10** [comment]
> 感谢你的贡献，请问能否将截图删除，换成文字表述，这样就不用管理图片了，也便于其他朋友检索。

ok, all right

**Comment by youyou-pm10** [comment]
It has been modified before, remember to merge,thanks

File: README.md [hunk 1]
```diff
@@ -50,6 +50,17 @@ options:
   --download-image  Download images to local
 ```
 
+## Tips
+
+Some wrong will ouccured if you have ever imported a unsanitizer markdwon file such as uploading failly.The console will prompt non-standard documents in red font, and you need to view them manually.
```
Comment (alswl) [inline README.md:55]: suggest:

If you have imported illegal Markdwon files (e.g. failed uploads), some error messages will appear. The console will indicate illegal files in red font and you will need to view them manually.

File: yuque2markdown.py [hunk 2]
```diff
@@ -16,7 +17,8 @@
 TYPE_TITLE = "TITLE"
 TYPE_DOC = "DOC"
 META_JSON = "$meta.json"
-TMP_DIR = "/tmp"
+# windows can't mkdir multi-level directory, empty tmp_dir can enhance compatibility
+TMP_DIR = ""
```
Comment (alswl) [inline yuque2markdown.py:21]: The original design was to execute it in the /tmp directory and leave it to the system to clean up the temporary files, but in the case of windows, you can use the Windows TEMP variable.

---

原来的设计是在 /tmp 目录下面执行，交给系统自己清理临时文件，考虑 windows 情况，可以使用 Windows 的 TEMP 变量。

https://stackoverflow.com/questions/847850/cross-platform-way-of-getting-temp-directory-in-python

File: yuque2markdown.py [hunk 3]
```diff
@@ -111,6 +118,10 @@ def download_images_and_patch_html(output_dir_path, sanitized_title, html):
         no = 1
         for image in bs.find_all("img"):
             print("Download %s" % image["src"])
+            # check the url whether is valid, sometimes we import a wrong md so alert it to fix by handwrite.
+            if not is_valid_url(image["src"]):
+                print("\033[31mDocx '%s' has a wrong img path, check it!\033[0m" % sanitized_title)
```
Comment (alswl) [inline yuque2markdown.py:123]: wrong -> invalid.

And please disable the Shell color, because some Shell env will not decode the prompts.

**Pull request closed without merging**